pandas/core/arrays/categorical.py

```diff
@@ -51,7 +52,7 @@
     _extension_array_shared_docs,
     try_cast_to_ea,
 )
-from pandas.core.base import NoNewAttributesMixin, PandasObject, _shared_docs
+from pandas.core.base import IndexOpsMixin, NoNewAttributesMixin, PandasObject
```

Comment:
I believe I am in the minority on this view and don't want to be overly difficult, but can you refactor the `IndexOpsMixin` as a pre-cursor to this, or leave this module separate from the rest of changes (which look good btw)

Reply:
Yes, that is a good idea. I would convert this back to using `_shared_docs`. This PR is already too large, and I should really put them separately.